lms/djangoapps/teams/views.py

```diff
@@ -803,8 +803,13 @@ def get(self, request):
         """GET /api/team/v0/team_membership"""
         specified_username_or_team = False
         username = None
-        valid_courses = None
         team_id = None
+        course_id = None
+        valid_courses = None
+
+        if 'course_id' in request.QUERY_PARAMS:
+            course_id = request.QUERY_PARAMS['course_id']
+            valid_courses = [CourseKey.from_string(course_id)]
```

Comment:
`CourseKey.from_string` may raise an `InvalidKeyError`. Catch it and return a 404 in that case.